pkg/chunk/cache/redis_cache.go

```diff
@@ -0,0 +1,244 @@
+package cache
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"time"
+
+	"github.com/cortexproject/cortex/pkg/util"
+	"github.com/go-kit/kit/log/level"
+	"github.com/gomodule/redigo/redis"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	instr "github.com/weaveworks/common/instrument"
+)
+
+// RedisClient interface exists for mocking redisClient.
+type RedisClient interface {
+	Connection() redis.Conn
+	Close() error
+}
+
+// RedisCache type caches chunks in redis
+type RedisCache struct {
+	name            string
+	expiration      int
+	timeout         time.Duration
+	client          RedisClient
+	requestDuration observableVecCollector
+}
+
+// RedisConfig defines how a RedisCache should be constructed.
+type RedisConfig struct {
+	Endpoint       string        `yaml:"endpoint,omitempty"`
+	Timeout        time.Duration `yaml:"timeout,omitempty"`
+	Expiration     time.Duration `yaml:"expiration,omitempty"`
+	MaxIdleConns   int           `yaml:"max_idle_conns,omitempty"`
+	MaxActiveConns int           `yaml:"max_active_conns,omitempty"`
+}
+
+type redisClient struct {
+	pool *redis.Pool
+}
+
+var (
+	redisRequestDuration = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: "cortex",
+		Name:      "redis_request_duration_seconds",
+		Help:      "Total time spent in seconds doing redis requests.",
+		// Redis requests are very quick: smallest bucket is 16us, biggest is 1s
+		Buckets: prometheus.ExponentialBuckets(0.000016, 4, 8),
```

Comment:
microseconds?

Reply:
Yes, same as for `memcached`, I'm using a histogram with 8 buckets: 16us, 64us, 256us, 1ms, 4ms, 16ms, 64ms, 256ms.
Indeed, 16us and 64us might be too short, but 256us looks reasonable to me.
I suggest I change it to `prometheus.ExponentialBuckets(0.00025, 4, 8)`. Would that make sense?